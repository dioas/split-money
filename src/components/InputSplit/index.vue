<template>
  <div class="input-split">
    <input v-on:keyup.enter="onEnter" :value='value' @input="$emit('input', $event.target.value)" placeholder="Type something"/>
  </div>
</template>

<script>
export default {
  name: 'input-Split',
  props: {
    value: String,
    onEnter: Function
  }
}
</script>

<style lang="less" scoped>
.input-split {
  display: inline-block;
  input {
    width: 400px;
    height: 20px;
    border-radius: 5px;
    padding: 5px 10px;
    border: solid 1px #1BD060;
    &:focus {
      outline: none;
      border: solid 1px #009424;
    }
  }
}
</style>
