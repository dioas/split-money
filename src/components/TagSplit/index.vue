<template>
  <div class="tag-split">
    <div :class="'tag-value ' + splitCurrency.coler">{{splitCurrency.count}} X Rp{{splitCurrency.currency}}</div>
  </div>
</template>

<script>
export default {
  name: 'tag-split',
  props: {
    splitCurrency: Object
  }
}
</script>

<style lang="less" scoped>
.tag-split {
  display: inline-block;
  .tag-value {
    display: inline-block;
    color: white;
    padding: 5px 10px;
    border-radius: 3px;
    &.red {
      background-color: #E92B2B;
    }
    &.blue {
      background-color: #00A4AD;
    }
    &.green {
      background-color: #00A838;
    }
    &.orange {
      background-color: #E1920F;
    }
    &.lightYellow {
      background-color: #FFBA37;
    }
    &.blueOld {
      background-color: #00A4AD;
    }
    &.Bigcoin {
      background-color: gray;
    }
    &.mediumCoin {
      background-color: #E2E2E2;
    }
    &.smallCoin {
      background-color: #424242;
    }
    &.sisa {
      background-color: white;
      color:black;
    }
  }
}
</style>
