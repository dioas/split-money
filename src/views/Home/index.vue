<template>
  <div class="wrapper-content">
    <div class="instruction">
      <h3>Split Money will help you to check and split your total money for give to someone. Just try it and you will see a Magic here. Example, you can type "Rp 575.000" or something.</h3>
    </div>
    <div class="input-value">
      <InputComponent :onEnter="generateValue" v-model="money"/>
      <BtnComponent msg="Generate" :onClick="generateValue"/>
    </div>
    <div v-if="errorMessage" class="error-message">
      Something Wrong with your Input Currency, Please input correct currency
    </div>
    <div class="wrapper-tag">
      <div class="content-tag" v-for="(totalMoney, key) in this.splitCurrency" :key="key">
        <TagComponent :splitCurrency="totalMoney"/>
      </div>
    </div>
  </div>
</template>

<script>
import InputComponent from '@/components/InputSplit'
import BtnComponent from '@/components/ButtonSplit'
import TagComponent from '@/components/TagSplit'
import { mapActions } from 'vuex'
export default {
  name: 'home',
  components: {
    InputComponent, BtnComponent, TagComponent
  },
  data () {
    return {
      money: '',
      splitCurrency: [],
      errorMessage: false
    }
  },
  methods: {
    ...mapActions([
      'setSplitCurrency'
    ]),
    splitMoneyHere (money) {
      let temp = 0
      let data = {}
      if (money >= 100000) {
        temp = Math.floor(money / 100000)
        data = {
          count: temp,
          coler: 'red',
          currency: 100000
        }
        this.splitCurrency.push(data)
        this.splitMoneyHere(money % 100000)
      } else if (money >= 50000) {
        temp = Math.floor(money / 50000)
        data = {
          count: temp,
          coler: 'blue',
          currency: 50000
        }
        this.splitCurrency.push(data)
        this.splitMoneyHere(money % 50000)
      } else if (money >= 20000) {
        temp = Math.floor(money / 20000)
        data = {
          count: temp,
          coler: 'green',
          currency: 20000
        }
        this.splitCurrency.push(data)
        this.splitMoneyHere(money % 20000)
      } else if (money >= 10000) {
        temp = Math.floor(money / 10000)
        data = {
          count: temp,
          coler: 'orange',
          currency: 10000
        }
        this.splitCurrency.push(data)
        this.splitMoneyHere(money % 10000)
      } else if (money >= 5000) {
        temp = Math.floor(money / 5000)
        data = {
          count: temp,
          coler: 'lightYellow',
          currency: 5000
        }
        this.splitCurrency.push(data)
        this.splitMoneyHere(money % 5000)
      } else if (money >= 1000) {
        temp = Math.floor(money / 1000)
        data = {
          count: temp,
          coler: 'blueOld',
          currency: 1000
        }
        this.splitCurrency.push(data)
        this.splitMoneyHere(money % 1000)
      } else if (money >= 500) {
        temp = Math.floor(money / 500)
        data = {
          count: temp,
          coler: 'Bigcoin',
          currency: 500
        }
        this.splitCurrency.push(data)
        this.splitMoneyHere(money % 500)
      } else if (money >= 100) {
        temp = Math.floor(money / 100)
        data = {
          count: temp,
          coler: 'mediumCoin',
          currency: 100
        }
        this.splitCurrency.push(data)
        this.splitMoneyHere(money % 100)
      } else if (money >= 50) {
        temp = Math.floor(money / 50)
        data = {
          count: temp,
          coler: 'smallCoin',
          currency: 50
        }
        this.splitCurrency.push(data)
        this.splitMoneyHere(money % 50)
      } else if (money > 0) {
        data = {
          count: 'Sisa',
          coler: 'sisa',
          currency: money
        }
        this.splitCurrency.push(data)
      }
    },
    generateValue () {
      var upperValue = this.money.toUpperCase()
      var validationChecked = this.validationGenerator(upperValue)
      if (validationChecked) {
        this.errorMessage = false
        this.splitCurrency = []
        var currencyNumber = upperValue.replace('RP', '')
        this.splitMoneyHere(currencyNumber.replace(/[^0-9]/g, ''))
      } else {
        this.errorMessage = true
      }
    },
    validationGenerator (value) {
      if (value) {
        var checkSymbol = value.replace('RP', '').split(' ')
        var checkCharacter = value.indexOf('RP')
        var checkValue = value.replace(/[^0-9]/g, '')
        var checkNaN = value.replace('RP', '')
        var desimalValue = value.split(',')
        let isWrongValue
        if (desimalValue[1] !== undefined) {
          if (desimalValue[1].length !== 2) {
            isWrongValue = true
          } else {
            isWrongValue = false
          }
        }
        if (checkSymbol.length > 2) {
          return false
        } else if (checkCharacter > 1) {
          return false
        } else if (checkValue === '') {
          return false
        } else if (isNaN(parseInt(checkNaN))) {
          return false
        } else if (isWrongValue) {
          return false
        } else {
          return true
        }
      }
    }
  }
}
</script>

<style lang="less" scoped>
.input-value {
  display: inline-block;
  width: 100%;
  text-align: center;
}
.wrapper-content {
  background-color: white;
  min-height: 100vh;
  .instruction {
    h3 {
      width: 800px;
      display: inline-block;
    }
  }
  .wrapper-tag {
    width: 900px;
    display: inline-block;
    padding: 20px;
  }
  .content-tag {
    display: inline-block;
    margin: 0 10px;
  }
}
.error-message {
  color:#E92B2B;
  padding: 10px 0;
}
</style>
