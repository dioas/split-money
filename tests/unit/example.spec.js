import { mount } from '@vue/test-utils'
import Home from '@/App.vue'
import HomeIndex from '@/views/Home'
import Button from '@/components/ButtonSplit'
import InputSplit from '@/components/InputSplit'
import TagSplit from '@/components/InputSplit'

test('renders correctly', () => {
  const wrapper = mount(Home)
  expect(wrapper.element).toMatchSnapshot()
})
describe('Component', () => {
  test('is a Vue instance', () => {
    const wrapper = mount(Home)
    expect(wrapper.isVueInstance()).toBeTruthy()
  })
})
describe('Component', () => {
  test('is a Vue instance', () => {
    const wrapper = mount(HomeIndex)
    expect(wrapper.isVueInstance()).toBeTruthy()
  })
})
test('renders correctly', () => {
  const wrapper = mount(Button)
  expect(wrapper.element).toMatchSnapshot()
})
describe('Component', () => {
  test('is a Vue instance', () => {
    const wrapper = mount(Button)
    expect(wrapper.isVueInstance()).toBeTruthy()
  })
})

test('renders correctly', () => {
  const wrapper = mount(Button)
  expect(wrapper.element).toMatchSnapshot()
})
describe('Component', () => {
  test('is a Vue instance', () => {
    const wrapper = mount(Button)
    expect(wrapper.isVueInstance()).toBeTruthy()
  })
})
test('renders correctly', () => {
  const wrapper = mount(InputSplit)
  expect(wrapper.element).toMatchSnapshot()
})
describe('Component', () => {
  test('is a Vue instance', () => {
    const wrapper = mount(InputSplit)
    expect(wrapper.isVueInstance()).toBeTruthy()
  })
})

test('renders correctly', () => {
  const wrapper = mount(InputSplit)
  expect(wrapper.element).toMatchSnapshot()
})
describe('Component', () => {
  test('is a Vue instance', () => {
    const wrapper = mount(InputSplit)
    expect(wrapper.isVueInstance()).toBeTruthy()
  })
})
test('renders correctly', () => {
  const wrapper = mount(TagSplit)
  expect(wrapper.element).toMatchSnapshot()
})
describe('Component', () => {
  test('is a Vue instance', () => {
    const wrapper = mount(TagSplit)
    expect(wrapper.isVueInstance()).toBeTruthy()
  })
})

test('renders correctly', () => {
  const wrapper = mount(TagSplit)
  expect(wrapper.element).toMatchSnapshot()
})
describe('Component', () => {
  test('is a Vue instance', () => {
    const wrapper = mount(TagSplit)
    expect(wrapper.isVueInstance()).toBeTruthy()
  })
})