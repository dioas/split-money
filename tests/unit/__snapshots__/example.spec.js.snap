// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  class="containers"
  id="app"
>
  <div
    class="paper"
  >
    <div
      id="nav"
    >
      <div
        class="header"
      >
        <h1>
          Split Money
        </h1>
      </div>
    </div>
     
    <router-view />
  </div>
</div>
`;

exports[`renders correctly 2`] = `
<div
  class="button-split"
>
  <button>
    
  </button>
</div>
`;

exports[`renders correctly 3`] = `
<div
  class="button-split"
>
  <button>
    
  </button>
</div>
`;

exports[`renders correctly 4`] = `
<div
  class="input-split"
>
  <input
    placeholder="Type something"
  />
</div>
`;

exports[`renders correctly 5`] = `
<div
  class="input-split"
>
  <input
    placeholder="Type something"
  />
</div>
`;

exports[`renders correctly 6`] = `
<div
  class="input-split"
>
  <input
    placeholder="Type something"
  />
</div>
`;

exports[`renders correctly 7`] = `
<div
  class="input-split"
>
  <input
    placeholder="Type something"
  />
</div>
`;

exports[`renders correctly 8`] = `
<div
  class="input-split"
>
  <input
    placeholder="Type something"
  />
</div>
`;
